File: src/test/java/be/hubau/sane/parser/ParserTest.java
package be.hubau.sane.parser;

import java.io.IOException;
import java.io.InputStream;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ParserTest {

    @Test
    public void parseStandardExampleFile() throws IOException, ParserException {
        InputStream inputStream = ParserTest.class.getResourceAsStream("/example.sane");
        Parser parser = new Parser();
        Rule rule = parser.parse("sane", inputStream);

        assertEquals(23, rule.rules.size());
    }
}

File: src/test/java/be/hubau/sane/parser/ParserTest.kt
package be.hubau.sane.parser

import org.junit.jupiter.api.Test
import java.io.InputStream

class ParserTest {

    @Test
    fun parseStandardExampleFile() {
        val inputStream: InputStream = this.javaClass.getResourceAsStream("/example.sane")
        val parser = Parser
        val rule: Rule = parser.parse(ruleName = "sane", inputStream = inputStream)

        print("Spelling: ${rule.spelling}")
    }
}